example/firmware-ariel-os/src/main.rs

```diff
@@ -0,0 +1,132 @@
+#![no_std]
+#![no_main]
+#![feature(impl_trait_in_assoc_type)]
+#![feature(used_with_arg)]
+
+use ariel_os::{
+    asynch::Spawner,
+    debug::log::info,
+    reexports::embassy_usb,
+    time::{Duration, Timer},
+    usb,
+};
+
+use postcard_rpc::{
+    define_dispatch,
+    header::VarHeader,
+    server::{
+        impls::embassy_usb_v0_4::{
+            dispatch_impl::{WireRxBuf, WireRxImpl, WireSpawnImpl, WireStorageNoUsb, WireTxImpl},
+            PacketBuffers,
+        },
+        Dispatch, Server,
+    },
+};
+use static_cell::ConstStaticCell;
+use workbook_icd::{PingEndpoint, ENDPOINT_LIST, TOPICS_IN_LIST, TOPICS_OUT_LIST};
+
+pub struct Context;
+
+type AppMutex = embassy_sync::blocking_mutex::raw::CriticalSectionRawMutex;
```

Comment:
On ariel, any application might run in isr, one-thread or multi-threaded mode, so I tend to default to the criticalsection mutex.